src/Http/Routing/src/EndpointRoutingMiddleware.cs

```diff
@@ -56,7 +55,13 @@ internal sealed class EndpointRoutingMiddleware
 
         public Task Invoke(HttpContext httpContext)
         {
-            var feature = new EndpointSelectorContext();
+            var feature = new EndpointSelectorContext(httpContext);
+
+            // There's already an endpoint, skip maching completely
+            if (feature.Endpoint != null)
+            {
+                return _next(httpContext);
```

Comment:
This seems like this case should be logged.

Reply:
Done